http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/ResponseHandler.java

```diff
@@ -386,7 +383,10 @@ private Throwable wrapException(Throwable originalCause) {
     private void notifyIfResponseNotCompleted(ChannelHandlerContext handlerCtx) {
         RequestContext requestCtx = handlerCtx.channel().attr(REQUEST_CONTEXT_KEY).get();
         boolean responseCompleted = handlerCtx.channel().attr(RESPONSE_COMPLETE_KEY).get();
-        if (!responseCompleted) {
+        boolean lastHttpContentReceived = handlerCtx.channel().attr(LAST_HTTP_CONTENT_RECEIVED_KEY).get();
+        handlerCtx.channel().attr(KEEP_ALIVE).set(false);
```

Comment:
Wouldn't this get set when we close the channel (change above)?

Reply:
Yes for most of the cases but there are certain edge cases it would not get set. For example, a response header does not contain `connection: close` (so we set the `KEEP_ALIVE` to true) but `channelInactive` is invoked after the content is buffered, and in this case, the "bad" channel might get picked up again in a new request once it's released in the pool before it's actually closed.